cmake/AddFdbTest.cmake

```diff
@@ -91,6 +91,9 @@ function(add_fdb_test)
   # if the value was undefined before, it will still be
   # undefined.
   math(EXPR assigned_id "${test_idx} - 1")
+  math(EXPR test_file_idx "${CURRENT_TEST_FILE_INDEX} + ${NUM_TEST_FILES}")
+  set(CURRENT_TEST_FILE_INDEX "${test_file_idx}" PARENT_SCOPE)
+  math(EXPR assigned_test_number "${test_file_idx} - ${NUM_TEST_FILES}")
```

Comment:
Basically same as above: instead of introducing a new variable, just change the math for `assigned_id`